In [47]:
import pandas as pd
import dask.dataframe as dd
from dask.diagnostics import ProgressBar


## 데이터 검토

In [49]:
# 367	건축물대장	층별개요 (2022년 07월)	671.04	2022-08-24

ddf = dd.read_parquet("floor")
ddf.head()


관리_건축물대장_PK                  대지_위치                도로명_대지_위치  건물_명 시군구_코드  \
0  29155-15130  광주광역시 남구 주월동 997-12번지     광주광역시 남구 군분로77번안길 23  None  29155   
1  29155-15130  광주광역시 남구 주월동 997-12번지     광주광역시 남구 군분로77번안길 23  None  29155   
2   30200-2142  대전광역시 유성구 교촌동 638-2번지   대전광역시 유성구 교촌로6번길 45-14  None  30200   
3   30200-2142  대전광역시 유성구 교촌동 638-2번지   대전광역시 유성구 교촌로6번길 45-14  None  30200   
4   30200-2142  대전광역시 유성구 교촌동 638-2번지   대전광역시 유성구 교촌로6번길 45-14  None  30200   

  법정동_코드 대지_구분_코드     번     지 특수지_명  ...   구조_코드_명    기타_구조 주_용도_코드 주_용도_코드_명  \
0  10600        0  0997  0012  None  ...      벽돌구조       세벽   03001       소매점   
1  10600        0  0997  0012  None  ...      벽돌구조       세벽   01001      단독주택   
2  10200        0  0638  0002  None  ...  철근콘크리트구조  철근콘크리트조   11199  기타아동관련시설   
3  10200        0  0638  0002  None  ...  철근콘크리트구조  철근콘크리트조   11199  기타아동관련시설   
4  10200        0  0638  0002  None  ...      벽돌구조      조적조   10999  기타교육연구시설   

         기타_용도   면적(㎡)  주_부속_구분_코드 주_부속_구분_코드_명 면적_제외_여부     생성_일자  
0           점포   69.69           0         주건축물     None  20100105  
1           주택   69.69           0         주건축물     None  20100105  
2         아동시설  177.95           0         주건축물     None  20100105  
3         아동시설  168.05           0         주건축물     None  20100105  
4  교육연구 및 복지시설  133.37           0         주건축물     None  20100105  

[5 rows x 33 columns]

In [50]:
# 세종시 한솔동 939 확인

df_hsdong939 = ddf[
    (ddf["시군구_코드"] == "36110") & (ddf["법정동_코드"] == "10600") & (ddf["번"] == "0939")
].compute()
df_hsdong939


관리_건축물대장_PK               대지_위치          도로명_대지_위치    건물_명 시군구_코드  \
243868  36110-100199613  세종특별자치시  한솔동 939번지   세종특별자치시  노을3로 14  첫마을아파트  36110   
88810   36110-100199545  세종특별자치시  한솔동 939번지   세종특별자치시  노을3로 14  첫마을아파트  36110   
88811   36110-100199545  세종특별자치시  한솔동 939번지   세종특별자치시  노을3로 14  첫마을아파트  36110   
88812   36110-100199545  세종특별자치시  한솔동 939번지   세종특별자치시  노을3로 14  첫마을아파트  36110   
88813   36110-100199545  세종특별자치시  한솔동 939번지   세종특별자치시  노을3로 14  첫마을아파트  36110   
...                 ...                 ...                ...     ...    ...   
39799   36110-100198907  세종특별자치시  한솔동 939번지   세종특별자치시  노을3로 14  첫마을아파트  36110   
39800   36110-100198907  세종특별자치시  한솔동 939번지   세종특별자치시  노을3로 14  첫마을아파트  36110   
39801   36110-100198907  세종특별자치시  한솔동 939번지   세종특별자치시  노을3로 14  첫마을아파트  36110   
39802   36110-100198907  세종특별자치시  한솔동 939번지   세종특별자치시  노을3로 14  첫마을아파트  36110   
39803   36110-100198907  세종특별자치시  한솔동 939번지   세종특별자치시  노을3로 14  첫마을아파트  36110   

       법정동_코드 대지_구분_코드     번     지 특수지_명  ...   구조_코드_명     기타_구조 주_용도_코드  \
243868  10600        0  0939  0000  None  ...  철근콘크리트구조  철근콘크리트구조   02001   
88810   10600        0  0939  0000  None  ...  철근콘크리트구조  철근콘크리트구조   02001   
88811   10600        0  0939  0000  None  ...  철근콘크리트구조  철근콘크리트구조   02001   
88812   10600        0  0939  0000  None  ...  철근콘크리트구조  철근콘크리트구조   02001   
88813   10600        0  0939  0000  None  ...  철근콘크리트구조  철근콘크리트구조   02001   
...       ...      ...   ...   ...   ...  ...       ...       ...     ...   
39799   10600        0  0939  0000  None  ...  철근콘크리트구조  철근콘크리트구조   02001   
39800   10600        0  0939  0000  None  ...  철근콘크리트구조  철근콘크리트구조   02001   
39801   10600        0  0939  0000  None  ...  철근콘크리트구조  철근콘크리트구조   02001   
39802   10600        0  0939  0000  None  ...  철근콘크리트구조  철근콘크리트구조   02001   
39803   10600        0  0939  0000  None  ...  철근콘크리트구조  철근콘크리트구조   02001   

       주_용도_코드_명           기타_용도    면적(㎡)  주_부속_구분_코드 주_부속_구분_코드_명 면적_제외_여부  \
243868       아파트  관리사무소,MDF실,방재실   264.40           1        부속건축물        0   
88810        아파트             아파트  1167.70           0         주건축물        0   
88811        아파트             아파트   663.36           0         주건축물        0   
88812        아파트             아파트   663.36           0         주건축물        0   
88813        아파트             아파트  1037.04           0         주건축물        0   
...          ...             ...      ...         ...          ...      ...   
39799        아파트             아파트   590.92           0         주건축물        0   
39800        아파트             아파트   590.92           0         주건축물        0   
39801        아파트             아파트   213.01           0         주건축물        0   
39802        아파트             아파트   270.00           0         주건축물        0   
39803        아파트             아파트   337.75           0         주건축물        0   

           생성_일자  
243868  20200428  
88810   20190603  
88811   20190603  
88812   20190603  
88813   20190603  
...          ...  
39799   20190603  
39800   20190603  
39801   20190603  
39802   20190603  
39803   20190603  

[236 rows x 33 columns]

In [51]:
len(df_hsdong939["관리_건축물대장_PK"].unique())


28

한솔동 939

건축물대장상 주건축물 12개(상가동 포함),


 부속건축물 16개(지하주차장 포함)

층별개요 관리_건축물대장_PK 28개로 동일

In [55]:
pd.DataFrame(df_hsdong939[["주_용도_코드", "주_용도_코드_명", "기타_용도"]].value_counts())


0
주_용도_코드 주_용도_코드_명   기타_용도                 
02001   아파트         아파트                189
                    설비배관공간              13
                    세대공용창고               5
                    계단실,승강기,홀            4
                    주민공동시설               4
                    경비실                  4
                    지하주차장                3
                    재활용품보관소              2
                    경로당                  2
                    보육시설                 1
                    문고                   1
                    독서실                  1
                    기계실,전기실,발전기실,열교환실    1
                    주민체육시설               1
                    관리사무소,MDF실,방재실       1
04999   기타제2종근린생활시설 계단실                  1
                    전기실,발전기실             1
                    제1,2종근린생활시설          1
                    주차장                  1

각 층별 용도를 보면,

 아파트 단지의 별개 부속건축물로 건축된 다양한 시설이 주 용도 코드와 이름에서는 모두 아파트 또는 근린생활시설로 표현되고 있음.

실제 현황을 파악하려면 기타_용도까지 함께 분석하여야 함. (기타 용도까지 "아파트"여야 주거)

In [57]:
# 서울시 관악구 신림동 610-84 확인

df_sldong61084 = ddf[
    (ddf["시군구_코드"] == "11620")
    & (ddf["법정동_코드"] == "10200")
    & (ddf["번"] == "0610")
    & (ddf["지"] == "0084")
].compute()
df_sldong61084


관리_건축물대장_PK                   대지_위치             도로명_대지_위치  건물_명 시군구_코드  \
35975  11620-30404  서울특별시 관악구 신림동 610-84번지   서울특별시 관악구 법원단지7길 28  성원빌라  11620   
35976  11620-30404  서울특별시 관악구 신림동 610-84번지   서울특별시 관악구 법원단지7길 28  성원빌라  11620   
35977  11620-30404  서울특별시 관악구 신림동 610-84번지   서울특별시 관악구 법원단지7길 28  성원빌라  11620   
35978  11620-30404  서울특별시 관악구 신림동 610-84번지   서울특별시 관악구 법원단지7길 28  성원빌라  11620   
35979  11620-30404  서울특별시 관악구 신림동 610-84번지   서울특별시 관악구 법원단지7길 28  성원빌라  11620   

      법정동_코드 대지_구분_코드     번     지 특수지_명  ...   구조_코드_명    기타_구조 주_용도_코드  \
35975  10200        0  0610  0084  None  ...  철근콘크리트구조  철근콘크리트조   02003   
35976  10200        0  0610  0084  None  ...  철근콘크리트구조  철근콘크리트조   02003   
35977  10200        0  0610  0084  None  ...  철근콘크리트구조  철근콘크리트조   02003   
35978  10200        0  0610  0084  None  ...  철근콘크리트구조  철근콘크리트조   02003   
35979  10200        0  0610  0084  None  ...  철근콘크리트구조  철근콘크리트조   02003   

      주_용도_코드_명       기타_용도   면적(㎡)  주_부속_구분_코드 주_부속_구분_코드_명 면적_제외_여부  \
35975     다세대주택  다세대주택(2세대)  112.77           0         주건축물     None   
35976     다세대주택  다세대주택(2세대)  132.54           0         주건축물     None   
35977     다세대주택  다세대주택(2세대)  134.79           0         주건축물     None   
35978     다세대주택  다세대주택(2세대)  134.79           0         주건축물     None   
35979     다세대주택  다세대주택(2세대)  134.79           0         주건축물     None   

          생성_일자  
35975  20171205  
35976  20171205  
35977  20171205  
35978  20171205  
35979  20171205  

[5 rows x 33 columns]

In [58]:
pd.DataFrame(df_sldong61084[["층_구분_코드", "층_구분_코드_명", "층_번호", "층_번호_명"]].value_counts())


0
층_구분_코드 층_구분_코드_명 층_번호 층_번호_명   
10      지하        1    지1층     1
20      지상        1    1층      1
                  2    2층      1
                  3    3층      1
                  4    4층      1

In [24]:
df_sldong61084[["주_용도_코드", "주_용도_코드_명", "기타_용도"]].value_counts()


주_용도_코드  주_용도_코드_명  기타_용도     
02003    다세대주택      다세대주택(2세대)    5
dtype: int64

반지하를 포함한 다세대 다가구 주택의 경우,
 기타 용도를 추가 정보를 기록하는 용도로 활용하는 경우가 있음.

확인한 사례: 다가구용단독주택(2가구),
 다가구주택(2가구),
 다세대주택(2세대) 등

## 전국 지하층 주택 규모 분석

주택 정의:

1. 주 용도 코드가 01 또는 02로 시작하고, 기타 용도에 주택, 다세대, 다가구, 공관, 아파트, 기숙사 등의 텍스트가 포함됨
2. 주 용도 코드가 01 또는 02로 시작하고, 기타 용도에 주택, 다세대, 다가구, 공관, 아파트, 기숙사 등의 텍스트가 포함되며, 설비, 배관, 창고, 계단, 승강기, 경비, 주차, 보관, 경로당, 보육, 문고, 독서, 기계, 전기, 체육, 사무소, MDF, 방재 등의 텍스트가 포함되지 않음

과대추정이 될 방식과 과소추정이 될 방식 두 가지로 주택 규모를 산출하여, 최종 결과의 오차 범위를 가늠함

지하층 정의: 층 구분 코드가 10임

반지하(층 바닥이 지면보다 낮지만 천장은 지면보다 높음) 추정은 불가능. 각 층의 바닥 및 천장 표고 (또는, 최고/최저 표고)가 건축물대장 또는 건축허브 등에 구축되어야 함.

반드시 지하 1층이 지면과 가까운 층인 것도 아님. 부산 등의 경우 지면에서 수평으로 지하 2층에 진입할 수 있는 경우 있음.

In [44]:
housing_contains = "|".join(["주택", "다세대", "다가구", "공관", "아파트", "기숙사"])
no_housing_contains = "|".join(
    [
        "설비",
        "배관",
        "창고",
        "계단",
        "승강기",
        "경비",
        "주차",
        "보관",
        "경로당",
        "보육",
        "문고",
        "독서",
        "기계",
        "전기",
        "체육",
        "사무소",
        "MDF",
        "방재",
    ]
)

ddf1 = ddf[
    (ddf["층_구분_코드"] == "10")
    & (ddf["주_용도_코드"].str.contains("^01|^02"))
    & (ddf["기타_용도"].str.contains(housing_contains))
]
ddf2 = ddf1[~ddf1["기타_용도"].str.contains(no_housing_contains)]


In [50]:
with ProgressBar():
    ddf1.to_parquet("underground_housing1")
    ddf2.to_parquet("underground_housing2")


[########################################] | 100% Completed | 57.52 s
[########################################] | 100% Completed | 76.48 s


### 1번 정의

In [51]:
df1 = pd.read_parquet("underground_housing1")
len(df1)


385859

In [53]:
len(df1["관리_건축물대장_PK"].unique())


377262

In [61]:
# 각 건물동별로 주택이 용도인 지하층 중,
# 면적이 가장 넓은 층의 면적을 해당 건물동의 반지하 주택 면적 규모로 추정

df1.groupby("관리_건축물대장_PK")["면적(㎡)"].max().sum()


35528939.764350004

1번 정의에 따른 반지하 주택 규모: 377,262동 35,528,940㎡ (평균 94㎡/동)

In [ ]:
(df1["관리_건축물대장_PK"].value_counts() >= 2).sum()


6139

In [ ]:
df1["관리_건축물대장_PK"].value_counts()


11650-100294490    24
46130-25507        24
46130-26402        21
11530-100194390    19
28185-100198380    18
                   ..
42150-100181850     1
30200-10870         1
11500-8590          1
11500-16863         1
11470-12425         1
Name: 관리_건축물대장_PK, Length: 377262, dtype: int64

In [ ]:
df1[df1["관리_건축물대장_PK"] == "11650-100294490"][
    ["주_용도_코드", "주_용도_코드_명", "기타_용도"]
].value_counts()


주_용도_코드  주_용도_코드_명  기타_용도                      
02001    아파트        도시형생활주택(계단실,복도,ELEV)           8
                    도시형생활주택(로비,기계실,전기실,실외기실,창고)    8
                    도시형생활주택(주차장)                   7
                    도시형생활주택(주민커뮤니티시설)              1
dtype: int64

예상했던 대로, 기타 용도에 주택이 들어있다고 모두 주택인 것이 아님.
지하층 "주택"이 2개층 이상 포함된 6,139동 중 이러한 오류가 얼마인지는 추가 연구가 필요한 부분.

한 건물동에 한 층의 면적만 세었기 때문에, 동 수와 바닥면적에 미치는 오차가 오류가 있는 층수만큼 증폭되지는 않음.

In [97]:
df1["주_용도_코드"].value_counts().head(20)


01001    150339
01003    122247
02003     72459
02001     20083
02002      9814
01002      7380
02005      2256
02007       879
02006       178
01000        86
02004        69
01004        45
02000        24
Name: 주_용도_코드, dtype: int64

1번 지하층 주택의 주 용도 코드는 단독주택, 다가구주택, 다세대주택, 아파트, 연립주택, 다중주택 순

### 2번 정의

In [78]:
df2 = pd.read_parquet("underground_housing2")
len(df2)


361081

In [79]:
len(df2["관리_건축물대장_PK"].unique())


358222

In [80]:
df2.groupby("관리_건축물대장_PK")["면적(㎡)"].max().sum()


29625848.811250005

2번 정의에 따른 반지하 주택 규모: 358,222동 29,625,849㎡ (평균 83㎡/동)

In [81]:
df2["관리_건축물대장_PK"].value_counts()


46130-25507        24
46130-26402        21
46130-29171        16
28185-100198380    13
11290-22653        10
                   ..
11380-35148         1
43113-124570        1
11200-6444          1
11290-37925         1
11470-12425         1
Name: 관리_건축물대장_PK, Length: 358222, dtype: int64

In [84]:
df2[df2["관리_건축물대장_PK"] == "46130-25507"][
    ["층_구분_코드", "층_구분_코드_명", "층_번호", "층_번호_명"] + ["주_용도_코드", "주_용도_코드_명", "기타_용도"]
].value_counts()


층_구분_코드  층_구분_코드_명  층_번호  층_번호_명  주_용도_코드  주_용도_코드_명  기타_용도   
10       지하         1     지1      01001    단독주택       주택-052      4
                                                      주택-054      4
                                  02007    기숙사        연립주택-042    3
                                  01001    단독주택       주택-055      2
                                                      주택022       2
                                  02007    기숙사        연립주택-033    2
                                  01001    단독주택       주택-051      1
                                                      주택-053      1
                                  02007    기숙사        연립주택-013    1
                                                      연립주택-023    1
                                                      연립주택-024    1
                                                      연립주택-026    1
                                                      연립주택-032    1
dtype: int64

In [85]:
df2[df2["관리_건축물대장_PK"] == "46130-25507"]


관리_건축물대장_PK               대지_위치          도로명_대지_위치  \
__null_dask_index__                                                       
212158               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212190               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212193               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212197               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212200               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212203               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212206               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212209               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212212               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212215               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212218               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212227               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212230               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212232               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212258               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212260               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212262               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212264               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212309               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212311               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212313               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212315               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212317               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   
212319               46130-25507  전라남도 여수시 안산동 290번지   전라남도 여수시 소호로 513   

                         건물_명 시군구_코드 법정동_코드 대지_구분_코드     번     지 특수지_명  ...  \
__null_dask_index__                                                     ...   
212158               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212190               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212193               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212197               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212200               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212203               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212206               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212209               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212212               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212215               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212218               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212227               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212230               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212232               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212258               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212260               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212262               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212264               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212309               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212311               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212313               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212315               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212317               한화석유화학사택  46130  13000        0  0290  0000  None  ...   
212319               한화석유화학사택  46130  13000        0  0290  0000  None  ...   

               

항공사진으로 확인하면 타운하우스 단지형 사택을 확인할 수 있음. 경사지에서 지면에서 진입하는 층이 지하층으로 세어짐. 단지 전체 중 한 동만 세어져 과소추정됨.

In [100]:
ddf2["주_용도_코드"].value_counts().head(20)


01001    140977
01003    119169
02003     68091
02001     14645
02002      8904
01002      7291
02005       928
02007       728
02006       133
01000        86
02004        64
01004        42
02000        23
Name: 주_용도_코드, dtype: int64

2번 지하층 주택의 주 용도 코드도 단독주택, 다가구주택, 다세대주택, 아파트, 연립주택, 다중주택 순

### 3번 정의

주택 정의 3. 주 용도 코드가 단독주택, 다중주택, 다가구주택, 연립주택, 다세대주택 중 하나이고, 기타 용도에 주택, 다세대, 다가구 등의 텍스트가 포함되며, 설비, 배관, 창고, 계단, 승강기, 경비, 주차, 보관, 경로당, 보육, 문고, 독서, 기계, 전기, 체육, 사무소, MDF, 방재 등의 텍스트가 포함되지 않음

아파트를 제외한 수치. (아파트는 반지하가 없다는 판단 또는 과소추정 유도)

In [104]:
# 단독, 다중, 다가구, 연립, 다세대만
narrower_housing = "|".join(
    [
        "01001",
        "01002",
        "01003",
        "02002",
        "02003",
    ]
)
narrower_housing_contains = "|".join(["주택", "다세대", "다가구"])

ddf3 = ddf[
    (ddf["층_구분_코드"] == "10")
    & (ddf["주_용도_코드"].str.contains(narrower_housing))
    & (ddf["기타_용도"].str.contains(narrower_housing_contains))
    & (~ddf1["기타_용도"].str.contains(no_housing_contains))
]

with ProgressBar():
    ddf3.to_parquet("underground_housing3")


[########################################] | 100% Completed | 71.94 s


In [3]:
df3 = pd.read_parquet("underground_housing3")
len(df3)


344315

In [107]:
len(df3["관리_건축물대장_PK"].unique())


342481

In [108]:
df3.groupby("관리_건축물대장_PK")["면적(㎡)"].max().sum()


22779915.635599997

3번 정의에 따른 반지하 주택 규모: 342,481동 22,779,916㎡ (평균 67㎡/동)

In [109]:
df3.head()


관리_건축물대장_PK                     대지_위치  \
__null_dask_index__                                          
50                   28237-22914   인천광역시 부평구 산곡동 369-495번지   
170                   44150-8790      충청남도 공주시 금학동 238-1번지   
176                  44150-27038  충청남도 공주시 반포면 봉곡리 180-2번지   
389                  11215-24296     서울특별시 광진구 화양동 44-54번지   
394                   11110-7401     서울특별시 종로구 부암동 240-1번지   

                                   도로명_대지_위치  건물_명 시군구_코드 법정동_코드 대지_구분_코드  \
__null_dask_index__                                                         
50                         인천광역시 부평구 마장로 147  None  28237  10300        0   
170                          충청남도 공주시 하선길 32  None  44150  11000        0   
176                     충청남도 공주시 반포면 원당1길 68  None  44150  34029        0   
389                   서울특별시 광진구 아차산로23길 13-4  None  11215  10700        0   
394                      서울특별시 종로구 백석동길 11-5  None  11110  18400        0   

                        번     지 특수지_명  ... 구조_코드_명 기타_구조 주_용도_코드 주_용도_코드_명  \
__null_dask_index__                    ...                                   
50                   0369  0495  None  ...    벽돌구조   연와조   01001      단독주택   
170                  0238  0001  None  ...    벽돌구조  세멘벽돌   01001      단독주택   
176                  0180  0002  None  ...    벽돌구조   연와조   01001      단독주택   
389                  0044  0054  None  ...    벽돌구조   연와조   01001      단독주택   
394                  0240  0001  None  ...    벽돌구조   연와조   01001      단독주택   

                    기타_용도  면적(㎡)  주_부속_구분_코드 주_부속_구분_코드_명 면적_제외_여부     생성_일자  
__null_dask_index__                                                           
50                     주택  76.02           0         주건축물     None  20110416  
170                    주택  42.34           0         주건축물     None  20110415  
176                    주택   8.91           0         주건축물     None  20110415  
389                    주택  68.58           0         주건축물     None  20170315  
394                    주택  30.88           0         주건축물     None  20170315  

[5 rows x 33 columns]

## 종합

|지하층 주택 정의|건축물 수(동)|바닥면적(m²)|
|-------------------------|-----:|-----:|
|지하층, 용도코드 주택, 기타 용도 주택 포함|377,262|35,528,940|
|지하층, 용도코드 주택, 기타 용도 주택 포함 및 타 용도 불포함|358,222|29,625,849|
|지하층, 용도코드 단독·다중·다가구·연립·다세대, 기타 용도 주택 포함 및 타 용도 불포함|342,481|22,779,916|

현재 건축물대장 데이터로는 반지하 층을 구별할 수 없으며, 지하층 주택의 정의에 따라 30~40만 동, 20~40백만m² 범위에서 반지하 주택 규모를 추정할 수 있음. 실제 반지하 여부와 세대 수 등은 중장기 연구과제를 통하여 추정하거나 전수조사 사업으로 확인하여야 하는 영역임.

## 지하층 주택 상위 지자체

3번 정의 사용. 시군구별 및 법정동별, 건물동 및 바닥면적 상위 지자체 목록.

In [13]:
df_sgg = pd.read_csv("code_sgg.csv", index_col="시군구코드", dtype={"시군구코드": str})
df_sgg


시군구명 폐지여부
시군구코드                   
11000         서울특별시   존재
11110     서울특별시 종로구   존재
11140      서울특별시 중구   존재
11170     서울특별시 용산구   존재
11200     서울특별시 성동구   존재
...             ...  ...
49710      제주도 북제주군   폐지
49720      제주도 남제주군   폐지
50000       제주특별자치도   존재
50110   제주특별자치도 제주시   존재
50130  제주특별자치도 서귀포시   존재

[461 rows x 2 columns]

In [16]:
top_sgg_building = (
    df3.groupby("시군구_코드")["관리_건축물대장_PK"].nunique().sort_values(ascending=False).head(20)
)
pd.DataFrame(top_sgg_building).join(df_sgg["시군구명"])


관리_건축물대장_PK        시군구명
시군구_코드                         
11620         15313   서울특별시 관악구
11260         12472   서울특별시 중랑구
11305         12382   서울특별시 강북구
11380         11878   서울특별시 은평구
11290         11853   서울특별시 성북구
11215         10598   서울특별시 광진구
11590          9401   서울특별시 동작구
11410          8325  서울특별시 서대문구
11560          8032  서울특별시 영등포구
11530          7370   서울특별시 구로구
41190          7201     경기도 부천시
11440          6951   서울특별시 마포구
11470          6860   서울특별시 양천구
11320          6518   서울특별시 도봉구
11200          6225   서울특별시 성동구
11740          6102   서울특별시 강동구
11545          5928   서울특별시 금천구
28237          5865   인천광역시 부평구
28177          5804  인천광역시 미추홀구
11710          5651   서울특별시 송파구

건축물 수 기준 서울 자치구가 상위 20개 시군구 중 17개에 해당하며, 그 외 부평시, 인천 부평구, 미추홀구 등이 상위에 포함됨

In [20]:
sgg_area = df3.groupby(["시군구_코드", "관리_건축물대장_PK"])["면적(㎡)"].max()
sgg_area


시군구_코드  관리_건축물대장_PK
11110   11110-10005     18.48
        11110-10006     70.23
        11110-10009     98.04
        11110-1001     145.06
        11110-10011    221.92
                        ...  
50130   50130-8404      19.38
        50130-9211     161.04
        50130-9212     161.04
        50130-9349       9.52
        50130-9739      15.66
Name: 면적(㎡), Length: 342481, dtype: float64

In [31]:
top_sgg_area = sgg_area.groupby("시군구_코드").sum().sort_values(ascending=False).head(20)

with pd.option_context("display.float_format", "{:,.0f}".format):
    display(pd.DataFrame(top_sgg_area).join(df_sgg["시군구명"]))


면적(㎡)         시군구명
시군구_코드                       
11620  1,068,716    서울특별시 관악구
11380    795,737    서울특별시 은평구
11260    745,727    서울특별시 중랑구
11215    715,405    서울특별시 광진구
11305    693,090    서울특별시 강북구
11290    669,933    서울특별시 성북구
11590    608,209    서울특별시 동작구
41190    537,887      경기도 부천시
11410    516,615   서울특별시 서대문구
11680    494,143    서울특별시 강남구
11710    482,966    서울특별시 송파구
11470    477,990    서울특별시 양천구
11530    468,944    서울특별시 구로구
11440    465,295    서울특별시 마포구
11560    456,648   서울특별시 영등포구
11740    451,377    서울특별시 강동구
11320    404,991    서울특별시 도봉구
41271    388,956  경기도 안산시 상록구
11500    386,575    서울특별시 강서구
11545    383,760    서울특별시 금천구

면적 기준 서울시 은평구, 부천시, 안산시 상록구 등이 더 높은 순위로 포함됨.

In [32]:
df_bjd = pd.read_csv("code_bjd.csv", index_col="법정동코드", dtype={"법정동코드": str})
df_bjd


법정동명 폐지여부
법정동코드                                
1100000000                 서울특별시   존재
1111000000             서울특별시 종로구   존재
1111010100         서울특별시 종로구 청운동   존재
1111010200         서울특별시 종로구 신교동   존재
1111010300         서울특별시 종로구 궁정동   존재
...                          ...  ...
5013032022  제주특별자치도 서귀포시 표선면 하천리   존재
5013032023  제주특별자치도 서귀포시 표선면 성읍리   존재
5013032024  제주특별자치도 서귀포시 표선면 가시리   존재
5013032025  제주특별자치도 서귀포시 표선면 세화리   존재
5013032026  제주특별자치도 서귀포시 표선면 토산리   존재

[46333 rows x 2 columns]

In [34]:
df3["법정동10"] = df3["시군구_코드"] + df3["법정동_코드"]
df3[["시군구_코드", "법정동_코드", "법정동10"]]


시군구_코드 법정동_코드       법정동10
__null_dask_index__                          
50                   28237  10300  2823710300
170                  44150  11000  4415011000
176                  44150  34029  4415034029
389                  11215  10700  1121510700
394                  11110  18400  1111018400
...                    ...    ...         ...
266474               11260  10500  1126010500
266478               11260  10500  1126010500
266485               47113  11900  4711311900
266494               11470  10300  1147010300
266631               11470  10300  1147010300

[344315 rows x 3 columns]

In [39]:
top_bjd_building = (
    df3.groupby(["법정동10"])["관리_건축물대장_PK"]
    .nunique()
    .sort_values(ascending=False)
    .head(20)
)
pd.DataFrame(top_bjd_building).join(df_bjd["법정동명"])


관리_건축물대장_PK            법정동명
법정동10                                  
1162010200         8895   서울특별시 관악구 신림동
1126010100         6043   서울특별시 중랑구 면목동
1162010100         6033   서울특별시 관악구 봉천동
1130510100         5416   서울특별시 강북구 미아동
1130510300         4574   서울특별시 강북구 수유동
1156013200         3875  서울특별시 영등포구 신길동
1150010300         3490   서울특별시 강서구 화곡동
1147010300         3347   서울특별시 양천구 신월동
1121510100         3318   서울특별시 광진구 중곡동
1129013800         3071   서울특별시 성북구 장위동
1154510200         2862   서울특별시 금천구 독산동
1121510500         2822   서울특별시 광진구 자양동
1154510300         2776   서울특별시 금천구 시흥동
1159010200         2643   서울특별시 동작구 상도동
1159010700         2635   서울특별시 동작구 사당동
1153010200         2546   서울특별시 구로구 구로동
1156013300         2507  서울특별시 영등포구 대림동
2817710500         2474  인천광역시 미추홀구 주안동
1138010300         2359   서울특별시 은평구 불광동
1174010900         2286   서울특별시 강동구 천호동

건축물 수 기준 서울 법정동이 상위 20개 법정동 중 19개에 해당하며, 그 외에는 인천 미추홀구 주안동이 해당됨.

In [40]:
bjd_area = df3.groupby(["법정동10", "관리_건축물대장_PK"])["면적(㎡)"].max()
bjd_area


법정동10       관리_건축물대장_PK    
1111010100  11110-100190942     87.19
            11110-100192518     42.51
            11110-100204258     40.61
            11110-100205367     65.08
            11110-100217265     36.25
                                ...  
5013032024  50130-100354443    108.26
5013032025  50130-100352605     34.65
            50130-100355383     34.65
5013032026  50130-100203196     36.19
            50130-100256375     18.14
Name: 면적(㎡), Length: 342481, dtype: float64

In [45]:
top_bjd_area = bjd_area.groupby("법정동10").sum().sort_values(ascending=False).head(20)

with pd.option_context("display.float_format", "{:,.0f}".format):
    display(pd.DataFrame(top_bjd_area).join(df_bjd["법정동명"]))


면적(㎡)            법정동명
법정동10                             
1162010200 630,593   서울특별시 관악구 신림동
1162010100 406,553   서울특별시 관악구 봉천동
1126010100 349,448   서울특별시 중랑구 면목동
1130510100 278,976   서울특별시 강북구 미아동
1130510300 269,295   서울특별시 강북구 수유동
1150010300 267,280   서울특별시 강서구 화곡동
1147010300 235,536   서울특별시 양천구 신월동
1121510100 221,977   서울특별시 광진구 중곡동
1156013200 214,022  서울특별시 영등포구 신길동
1154510200 191,838   서울특별시 금천구 독산동
1121510500 179,476   서울특별시 광진구 자양동
1159010200 178,318   서울특별시 동작구 상도동
1154510300 175,668   서울특별시 금천구 시흥동
4121010100 165,922     경기도 광명시 광명동
1159010700 162,905   서울특별시 동작구 사당동
1138010300 159,610   서울특별시 은평구 불광동
1174010900 153,694   서울특별시 강동구 천호동
1156013300 150,631  서울특별시 영등포구 대림동
1129013800 146,646   서울특별시 성북구 장위동
2817710500 142,474  인천광역시 미추홀구 주안동

면적 기준 광명시 광명동이 추가로 포함됨